src/MerchantCenter/MerchantCenterService.php

```diff
@@ -334,4 +340,43 @@ protected function is_mc_contact_information_setup(): bool {
 
 		return $is_setup['phone_number'] && $is_setup['address'];
 	}
+	
+	/*
+	 * Check if the taxes + shipping rate and time + free shipping settings have been saved.
+	 *
+	 * @return bool If all required settings have been provided.
+	 *
+	 * @since x.x.x
+	 */
+	protected function saved_shipping_and_tax_options(): bool {
+		$merchant_center_settings = $this->options->get( OptionsInterface::MERCHANT_CENTER, [] );
+		$target_countries         = $this->get_target_countries();
+
+		// Tax options saved if: not US (no taxes) or tax_rate has been set
+		if ( in_array( 'US', $target_countries, true ) && empty( $merchant_center_settings['tax_rate'] ) ) {
+			return false;
+		}
+
+		// Free shipping saved if: not offered, OR offered and threshold not null
+		if ( ! empty( $merchant_center_settings['offers_free_shipping'] ) && is_null( $merchant_center_settings['free_shipping_threshold'] ) ) {
```

Comment:
Shouldn't we check if `$merchant_center_settings['free_shipping_threshold']` isset?

Reply:
I _believe_ it's always `set` as soon as any MC setting is saved, and since there's no default value in the schema, it's null:
https://github.com/woocommerce/google-listings-and-ads/blob/9449ae0c20682040046ae36ad83b58a086edc73d/src/API/Site/Controllers/MerchantCenter/SettingsController.php#L69-L71

So if `offers_free_shipping` isn't empty (has been saved), then `free_shipping_threshold` must be `null` or have a value.

Reply:
Hmm. I'd say let's just [be defensive](https://en.wikipedia.org/wiki/Defensive_programming) here and make sure because there's no contract to enforce that in the code (like a value object). 

So in theory the `is_contact_information_setup` method could be called anywhere even when the `get_settings_endpoint_edit_callback` API method isn't used. Since it's public (say if we need it in another place in the future).

Reply:
Sounds like a plan! 🛡️ eed32930b478b9d203a0aaf67d7c5741a81bcb4b